app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidateEmailUseCase.kt

```diff
@@ -0,0 +1,18 @@
+package com.davidtomas.taskyapp.features.auth.domain.useCase
+
+import com.davidtomas.taskyapp.features.auth.domain.model.EmailFormatError
+import com.davidtomas.taskyapp.features.auth.domain.model.LoginInputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.MissingMandatoryField
+
+
+class ValidateEmailUseCase {
+    operator fun invoke(email: String): LoginInputValidationError? =
+        when {
+            email.isBlank() -> MissingMandatoryField
+            !email.matches(EMAIL_REGEX.toRegex()) -> EmailFormatError
+            else -> null
+        }
+    companion object {
+        const val EMAIL_REGEX = "[^@ \\t\\r\\n]+@[^@ \\t\\r\\n]+\\.[^@ \\t\\r\\n]+"
+    }
+}
```

Comment:
The safest for email validation is just using the built-in email regex: 
```
Patterns.EMAIL_ADDRESS.matcher(email).matches()
```
It's an Android import though, so using it in domain would require an abstraction. But I think that's worth it, since email regex can be incredibly complex and I'm pretty sure yours doesn't cover all types of emails